pkg/network/protocols/events/consumer.go

```diff
@@ -206,10 +215,15 @@ func (c *Consumer[V]) process(cpu int, b *batch, syncing bool) {
 		return
 	}
 
-	// telemetry stuff
-	c.batchSize.Store(int64(b.Cap))
 	c.eventsCount.Add(int64(end - begin))
-	c.kernelDropsCount.Add(int64(b.Dropped_events))
+
+	// Only track these bits of telemetry when we're reading the batch off the
```

Comment:
Could you please provide more details on why, in the Sync() flow, there will be a duplication of telemetry counts?

Reply:
calling `Sync()` will "pre-fetch" incomplete batches before they get flushed to the perf ring.

So let's say we have a batch like
```
idx: 5
len: 3
cap: 15
dropped_events: X
```
When we call `Sync()`.

Eventually this batch will be flushed in the perf ring and will look like this:
```
idx: 5
len: 15
cap: 15
dropped_events: Y
```

If we had no guard clause like `if !syncing` we would account twice for the telemetry and would emit the metric as `X+Y`, but what we actually want is the "final" number `Y`.